plugins/figura_format/figura_format.js

```diff
@@ -147,266 +115,339 @@
-										let values = {};
-										keys.forEach(key => values[key.hashCode()] = (index == 0));
-										dialog.setFormValues(values);
 									}
+									Undo.initEdit({ animations: animsToRemove })
+									if (form_result.replace_animations)
+										animsToRemove.forEach(anim => anim.remove(false))
+									let new_animations = Animator.loadFile({ json: animJson, path: null }, names);
+									Undo.finishEdit('Figura Import animations', { animations: new_animations })
+								},
+								onCancel() {
 								}
-								form.properties_break = '_'
-								form.replace_animations = {
-									label: "Replace Animations?",
-									description: "If enabled, the imported animations will replace the old ones",
-									type: 'checkbox',
-									value: true,
-									nocolon: true
-								};
-								dialog.show();
-							}
-						})
-					}
-				}), '#file')
-			MenuBar.menus.animation.addAction(
-				new Action('figura_bake_ik', {
-					name: "Bake IK into Animations",
-					description: "Bakes Inverse Kinematics into raw Keyframes for use in Figura",
-					icon: "fa-bone",
-					condition: { modes: ['animate'], method: () => Format === format && Animation.selected },
-					click() {
-						new Dialog({
-							id: "figura_confirm_bake_ik",
-							title: "Confirm Bake Inverse Kinematics",
-							lines: [
-								"<p>This bakes the IK of all NullObjects onto the keyframes of the groups themselves, allowing it to be visible in Figura</p>",
-								"<p>However, NullObjects <strong>override</strong> the keyframes of the affected groups while it is present</p>",
-								"<p>Leaving the NullObject in the model has no effect on Figura, so just leave it incase you want to rebake the IK later</p>"
-							],
-							form: {
-								"all_animations": {
-									type: 'checkbox',
-									label: 'Bake all Animations?',
-									description: "This Action will normally only bake the selected Animation. Do you want to bake all Animations in one swoop?",
-									value: false,
-									full_width: false
+							});
+							form.select_all_none = {
+								type: 'buttons',
+								buttons: ['generic.select_all', 'generic.select_none'],
+								click(index) {
+									let values = {};
+									keys.forEach(key => values[key.hashCode()] = (index == 0));
+									dialog.setFormValues(values);
 								}
-							},
-							onConfirm() {
-								const animations = this.getFormResult().all_animations ? Animator.animations : [Animation.selected]
-								for (const animation of animations) {
-									let animators = animation.animators
+							}
+							form.properties_break = '_'
+							form.replace_animations = {
+								label: "Replace Animations?",
+								description: "If enabled, the imported animations will replace the old ones",
+								type: 'checkbox',
+								value: true,
+								nocolon: true
+							};
+							dialog.show();
+						}
+					})
+				}
+			})
+			const bakeIK = new Action('figura_bake_ik', {
```

Comment:
This was already in the plugin before this PR, but this seems to duplicate the built-in Bake Inverse Kinematics feature, except no Undo point is being created. Can we remove this?

Reply:
Yep. Removed